File: backend/src/main/java/com/vn/sodu/ui/service/StaticPageService.java
package com.vn.sodu.ui.service;

import com.vn.sodu.global.dto.PageResponse;
import com.vn.sodu.global.dto.SearchRequest;
import com.vn.sodu.ui.dto.StaticPageDTO;
import com.vn.sodu.ui.dto.StaticPageRequest;

public interface StaticPageService {
    StaticPageDTO createPage(StaticPageRequest request);
    StaticPageDTO updatePage(Long id, StaticPageRequest request);
    StaticPageDTO getPageById(Long id);
    StaticPageDTO getPublishedPageBySlug(String slug);
    PageResponse<StaticPageDTO> searchPages(SearchRequest request);
    void deletePage(Long id);
}

File: frontend/src/service/static-page.service.ts
import apiClient from '../api/api-client';
import {PageResponse} from '../interface/api-response';
import {StaticPageDTO, StaticPageMutationPayload} from '../interface/static-page.model';
import {UiSearchParams} from '../interface/public-ui-config.model';

export const StaticPageService = {
    searchPages: (searchTerm: string, params?: UiSearchParams): Promise<PageResponse<StaticPageDTO>> => {
        return apiClient.post('/api/admin/static-pages/search', {searchTerm, ...params});
    },

    getPageDetail: (id: string | number): Promise<StaticPageDTO> => {
        return apiClient.get(`/api/admin/static-pages/${id}`);
    },

    createPage: (data: StaticPageMutationPayload): Promise<StaticPageDTO> => {
        return apiClient.post('/api/admin/static-pages', data);
    },

    updatePage: (id: string | number, data: StaticPageMutationPayload): Promise<StaticPageDTO> => {
        return apiClient.put(`/api/admin/static-pages/${id}`, data);
    },

    deletePage: (id: string | number): Promise<void> => {
        return apiClient.delete(`/api/admin/static-pages/${id}`);
    },

    getPublishedPageBySlug: (slug: string): Promise<StaticPageDTO> => {
        return apiClient.get(`/api/public/pages/${slug}`);
    },
};
